src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -6211,20 +6211,29 @@ void func_8083C148(Player* this, PlayState* play) {
     this->stateFlags1 &= ~(PLAYER_STATE1_13 | PLAYER_STATE1_14 | PLAYER_STATE1_20);
 }
 
-s32 Player_ActionHandler_6(Player* this, PlayState* play) {
+/**
+ * Handles setting up a roll if it is appropriate.
+ *
+ * If a roll is not applicable, there are two extra behaviors that could occur:
+ * - If an item is currently held in hand, it can be put away.
+ * - Navi can be toggled on and off. She will either appear and fly around Link's head, or fly back into his body.
```

Comment:
into his hat, not his body, right?

Reply:
idk I don't want to get caught up in the lore, but "into him" or "into his body" makes it seem like navi "becomes one with link" or something, but I've always thought of fairies as a physical beings (navi bonks on a fence in the into for example). So when navi is put away she must be hiding in his hat or something, not in his body

Reply:
shes a magical being and is either phasing into links body or phasing into his hat haha, theres no difference to me

Reply:
I think I have never ever considered where do the companions of each Link go when they are not on screen.

Googling this specifically is very annoying since it yields a lot of stuff related to Navi leaving Link at the end of the game, but the [Zelda wiki](https://zelda.fandom.com/wiki/Navi) does agree that Navi hides into his hat
> When Navi's help is not being used, she seemingly hides herself inside Link's hat. However, she can be summoned at any time by pressing the A Button while Link is standing still (she will return to his hat if remaining still for 5 seconds).

After checking this by myself I have to agree it seems like she goes into Link's head/hat when hiding.

Even after this unexpected rabbithole, I'm not really sure if it would make a difference or be worth to mention it here since the main point of this comment is explaining that Navi's toggling is handled by this function.

Reply:
Yeah I dont disagree that she goes to his hat. I just see it as her phasing into it, and therefore dont see a problem with saying she "goes into him". But yeah agreed that it doesnt matter too much in this context haha

Reply:
whatever ill just be specific anyway :D have to commit again because of formatter complaints anyway